ui/utils.ts

```diff
@@ -0,0 +1,278 @@
+// If CALLBACK is given then it should be a callback
+// to call when close is pressed
+export function showError(title: string, content?: string, callback?: any) {
+    if (!callback) {
+        callback = function() {};
+    }
+    $.confirm({
+        title: title,
+        content: content,
+        type: 'red',
+        typeAnimated: true,
+        buttons: {
+            close: callback
+        }
+    });
+}
+
+// Show an Info message with TITLE and CONTENT
+export function showInfo(title: string, content: string) {
+    $.confirm({
+        title: title,
+        content: content,
+        type: 'green',
+        typeAnimated: true,
+        buttons: {
+            close: function() {}
+        }
+    });
+}
+
+export function showWarning(title: string, content: string) {
+    $.confirm({
+        title: title,
+        content: content,
+        type: 'yellow',
+        typeAnimated: true,
+        buttons: {
+            close: function() {
+                return;
+            }
+        }
+    });
+}
+
+// TODO: Remove this/replace with something else. In the case of a huge log hangs the entire app
+export function setup_log_watcher(callback: any) {
+    // if the log file is not found keep trying until it is
+    if (!fs.existsSync('rotkehlchen.log')) {
+        log_searcher = setInterval(function() {_setup_log_watcher(callback); }, 5000);
+        return;
+    }
+    _setup_log_watcher(callback);
+}
+
+export function setup_client_auditor() {
+    if (!client_auditor) {
+        client_auditor = setInterval(periodic_client_query, 60000);
+    }
+}
+
+export function reload_table_currency_val(table: any, colnum: any) {
+    table.rows().invalidate();
+    $(table.column(colnum).header()).text(settings.main_currency.ticker_symbol + ' value');
+    table.draw();
+}
+
+export function reload_table_currency_val_if_existing(table: any, colnum: any) {
+    if (table) {
+        reload_table_currency_val(table, colnum);
+    }
+}
+
+export function string_capitalize(s: string) {
+    return s && s[0].toUpperCase() + s.slice(1);
+}
+
+export function throw_with_trace(str: string) {
+    let err = new Error();
+    throw new Error(str + '\n' + err.stack);
+}
+
+export function date_text_to_utc_ts(txt: string | number | string[]) {
+    // for now assuming YYY/MM/DD HH:MM
+    if (settings.datetime_format !== 'd/m/Y G:i') {
+        throw new Error('Invalid datetime format');
+    }
+    let m = txt.toString().match(/\d+/g);
+    let day = parseInt(m[0], 10);
+    let month = parseInt(m[1], 10) - 1;
+    let year = parseInt(m[2], 10);
+    let hours = parseInt(m[3], 10);
+    let seconds = parseInt(m[4], 10);
+    return (new Date(Date.UTC(year, month, day, hours, seconds))).getTime() / 1000;
+}
+
+interface IctxMenuItems {
+    edit: any;
+    delete: any;
+    sep1: string;
+    quit: any;
+}
+
+export function dt_edit_drawcallback(id: string, edit_fn: any, delete_fn: any) {
+    return function (settings: any) {
```

Comment:
shouldn't settings be an object here?

Reply:
What type of object, what properties does it have? It's an unused variable, should it just be removed?